src/nbl/asset/utils/CHLSLCompiler.cpp

```diff
@@ -358,29 +360,46 @@ std::string CHLSLCompiler::preprocessShader(std::string&& code, IShader::E_SHADE
     if(context.get_hooks().m_pragmaStage != IShader::ESS_UNKNOWN)
         stage = context.get_hooks().m_pragmaStage;
 
-
-
+    // Context does not have access to the system, so it's up to us to add the lastWriteTime now
```

Comment:
you could just pass `m_system.get()` to the `context`'s constructor though, right?